src/UnionSet.jl

```diff
@@ -0,0 +1,258 @@
+"""
+function ρ(d::AbstractVector{N}, cap::UnionSet{N}) where {N<:Real}
+    X, Y = cap.X, cap.Y
+    return max(ρ(d, X), ρ(d, Y))
+end
+
+# ========================================
+# n-ary set union
+# ========================================
+
+"""
+    UnionSetArray{N<:Real, S<:LazySet{N}}
+
+Type that represents the set union of a finite number of convex sets.
+
+### Fields
+
+- `array` -- array of convex sets
+"""
+struct UnionSetArray{N<:Real, S<:LazySet{N}}
+    array::Vector{S}
+end
+
+# add functions connecting UnionSet and UnionSetArray
+@declare_array_version(UnionSet, UnionSetArray)
+
+"""
+    dim(cup::UnionSetArray)::Int
+
+Return the dimension of the set union of a finite number of convex sets.
+
+### Input
+
+- `cup` -- union of a finite number of convex sets
+
+### Output
+
+The ambient dimension of the union of a finite number of convex sets.
+"""
+function dim(cup::UnionSetArray)::Int
+    return dim(cup.array[1])
+end
+
+"""
+    array(cap::UnionSetArray{N, S})::Vector{S} where {N<:Real, S<:LazySet{N}}
+
+Return the array of a union of a finite number of convex sets.
+
+### Input
+
+- `cap` -- union of a finite number of convex sets
+
+### Output
+
+The array that holds the union of a finite number of convex sets.
+"""
+function array(cap::UnionSetArray{N, S})::Vector{S} where {N<:Real, S<:LazySet{N}}
+    return cap.array
+end
+
+"""
+    σ(d::AbstractVector{N}, cap::UnionSetArray{N}; algorithm="support_vector") where {N<:Real}
+
+Return the support vector of the union of a finite number of convex sets in
+a given direction.
+
+### Input
+
+- `d`         -- direction
+- `cap`       -- union of a finite number of convex sets
+- `algorithm` -- (optional, default: "support_vector"): the algorithm to compute
+                 the support vector; if "support_vector", use the support
+                 vector of each argument; if "support_function" use the support
+                 function of each argument and evalute the support vector of only
+                 one of them
+
+### Output
+
+The support vector in the given direction.
+
+### Algorithm
+
+The support vector of the union of a finite number of convex sets ``X₁, X₂, ...``
+can be obtained as the vector that maximizes the support function, i.e.
+it is sufficient to find the ``\\argmax(ρ(d, X₂), ρ(d, X₂), ...])`` and evaluate
+its support vector.
+
+The default implementation, with option `algorithm="support_vector"`, computes
+the support vector of all ``X₁, X₂, ...`` and then compares the support function using
+a dot product. If it happens that the support function can be more efficiently
+computed (without passing through the support vector), consider using the alternative
+`algorithm="support_function"` implementation, which evaluates the support function
+of each set directly and then calls only the support vector of only one of the ``Xᵢ``.
+"""
+function σ(d::AbstractVector{N}, cap::UnionSetArray{N}; algorithm="support_vector") where {N<:Real}
```

Comment:
These algorithms can be implemented in constant memory, but let us keep yours for simplicity.

Reply:
Do open an issue if you have an idea for improvement of this one, to modify `"support_function`, `"support_vector"` or add a new one.

Reply:
See #993.